kubernetes/samples/scripts/domain-lifecycle/startCluster.sh

```diff
@@ -0,0 +1,121 @@
+# !/bin/sh
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+source ${scriptDir}/helper.sh
+
+function usage() {
+
+  cat << EOF
+
+  This script starts a WebLogic cluster in a domain by patching
+  'spec.clusters[<cluster-name>].serverStartPolicy' attribute of the domain
+  resource to 'IF_NEEDED'. This change will cause the operator to initiate
+  startup of cluster's WebLogic server instance pods if the pods are not
+  already running and the spec.replicas or
+  'spec.clusters[<cluster-name>].serverStartPolicy' is set higher than zero.
+ 
+  Usage:
+ 
+    $(basename $0) -c mycluster [-n mynamespace] [-d mydomainuid] [-m kubecli]
+  
+    -c <cluster-name>   : Cluster name parameter is required.
+
+    -d <domain_uid>     : Domain unique-id. Default is 'sample-domain1'.
+
+    -n <namespace>      : Domain namespace. Default is 'sample-domain1-ns'.
+
+    -m <kubernetes_cli> : Kubernetes command line interface. Default is 'kubectl' if KUBERNETES_CLI env
+                          variable is not set. Otherwise default is the value of KUBERNETES_CLI env variable.
+
+    -h                  : This help.
+   
+EOF
+exit $1
+}
+
+set -eu
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+
+while getopts "c:n:m:d:h" opt; do
+  case $opt in
+    c) clusterName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+#
+# Function to perform validations, read files and initialize workspace
+#
+function initialize {
+
+  # Validate the required files exist
+  validateErrors=false
+
+  validateKubernetesCliAvailable
+  validateJqAvailable
+
+  if [ -z "${clusterName}" ]; then
+    validationError "Please specify cluster name using '-c' parameter e.g. '-c cluster-1'."
+  fi
+
+  isValidCluster=""
+  validateClusterName "${domainUid}" "${domainNamespace}" "${clusterName}" isValidCluster
+
+  if [ "${isValidCluster}" != 'true' ]; then
+    validationError "cluster ${clusterName} is not part of domain ${domainUid} in namespace ${domainNamespace}."
+  fi
+
+  failIfValidationErrors
+}
+
+initialize
+
+# Get the domain in json format
+domainJson=$(${kubernetesCli} get domain ${domainUid} -n ${domainNamespace} -o json)
+
+# Get server start policy for this cluster
```

Comment:
This could be simplified by leveraging the getClusterPolicy and getDomainPolicy helper.sh methods.